src/wallet/wallet.cpp

```diff
@@ -3711,18 +3711,10 @@ static int64_t GetOldestKeyTimeInPool(const std::set<int64_t>& setKeyPool, Walle
 int64_t CWallet::GetOldestKeyPoolTime()
 {
     LOCK(cs_wallet);
-
-    WalletBatch batch(*database);
-
-    // load oldest key from keypool, get time and return
-    int64_t oldestKey = GetOldestKeyTimeInPool(setExternalKeyPool, batch);
-    if (IsHDEnabled() && CanSupportFeature(FEATURE_HD_SPLIT)) {
-        oldestKey = std::max(GetOldestKeyTimeInPool(setInternalKeyPool, batch), oldestKey);
-        if (!set_pre_split_keypool.empty()) {
-            oldestKey = std::max(GetOldestKeyTimeInPool(set_pre_split_keypool, batch), oldestKey);
-        }
+    int64_t oldestKey = 0;
+    for (const auto& spk_man_pair : m_spk_managers) {
+        oldestKey = std::max(oldestKey, spk_man_pair.second->GetOldestKeyPoolTime());
```

Comment:
In d3dde67e4aba051bea38daae60ed80fd4db69d3b:  `GetOldestKeyPoolTime` was introduced in #10235. It calls `GetOldestKeyTimeInPool` for the internal and external keypools. IIUC this gets the timestamp from keypool entry at index 0. For legacy wallets those are the same, so `std::max` is fine. But if we add descriptor wallets, and allow importing them after wallet creation, then `std::max` would return the newest one. So maybe we should switch to `std::min` here and initialize `oldestKey` with `Time()`?  

Alternatively we could drop `keypoololdest` in the RPC wallet for non-legacy spk_managers, as it's useless for descriptor wallets anyway.

Reply:
I've changed it to use `min`. Looking at the original implementation in master, it actually looks like it's broken as it uses `max`.